File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n==0){
            return 0;
        }
        int first = 0;
        int last = 1;
        for(int i=1;i<n;i++){
            int sum = first + last;
            first = last;
            last = sum;
        }
        return last;
    }
}

File: 0509-fibonacci-number/0509-fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    if(n == 0) return 0;
    let first = 0;
    let last = 1;
    for(let i=1;i<n;i++){
        let sum = first+last;
        first = last;
        last = sum;
    }
    return last;
};